hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/db/RDBTable.java

```diff
@@ -125,8 +127,17 @@ public boolean isExist(byte[] key) throws IOException {
       // RocksDB#keyMayExist
       // If the key definitely does not exist in the database, then this
       // method returns false, else true.
-      return db.keyMayExist(handle, key, new StringBuilder())
-          && db.get(handle, key) != null;
+      rdbMetrics.incNumDBKeyMayExistChecks();
+      boolean keyMayExist = db.keyMayExist(handle, key, new StringBuilder());
+      if (!keyMayExist) {
```

Comment:
if (keyMayExist) {
boolean keyNotExist = (db.get(handle, key) == null);
if (keyNotExist) {
   rdbMetrics.incNumDBKeyMayExistMisses();
 }
  return keyNotExist;
} 
return false;
Can we modify the logic as above.